plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,170 @@
+package supervisor
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+	"net/url"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string `toml:"url"`
+	PidGather      bool   `toml:"gather_pid"`
+	ExitCodeGather bool   `toml:"gather_exit_code"`
+	UseIdentTag    bool   `toml:"use_identification_tag"`
+
+	Status SupervisorInfo
+}
+
+type ProcessInfo struct {
+	Name          string `xmlrpc:"name"`
+	Group         string `xmlrpc:"group"`
+	Description   string `xmlrpc:"description"`
+	Start         int32  `xmlrpc:"start"`
+	Stop          int32  `xmlrpc:"stop"`
+	Now           int32  `xmlrpc:"now"`
+	State         int16  `xmlrpc:"state"`
+	Statename     string `xmlrpc:"statename"`
+	StdoutLogfile string `xmlrpc:"stdout_logfile"`
+	StderrLogfile string `xmlrpc:"stderr_logfile"`
+	SpawnErr      string `xmlrpc:"spawnerr"`
+	ExitStatus    int8   `xmlrpc:"exitstatus"`
+	Pid           int32  `xmlrpc:"pid"`
+}
+
+type SupervisorInfo struct {
```

Comment:
I think we do not need to export this structure.
```suggestion
type supervisorInfo struct {
```

Reply:
Will be included in next commit